ngraph/core/src/op/proposal.cpp

```diff
@@ -37,51 +37,78 @@ op::v0::Proposal::Proposal(const Output<Node>& class_probs,
 void op::v0::Proposal::validate_and_infer_types()
 {
     NGRAPH_OP_SCOPE(v0_Proposal_validate_and_infer_types);
-    const auto& class_probs_pshape = get_input_partial_shape(0);
-    const auto& class_bbox_deltas_pshape = get_input_partial_shape(1);
-    const auto& image_shape_pshape = get_input_partial_shape(2);
-    if (class_probs_pshape.is_static() && class_bbox_deltas_pshape.is_static() &&
-        image_shape_pshape.is_static())
-    {
-        const Shape class_probs_shape{class_probs_pshape.to_shape()};
-        const Shape class_bbox_deltas_shape{class_bbox_deltas_pshape.to_shape()};
-        const Shape image_shape_shape{image_shape_pshape.to_shape()};
+    const auto& class_probs_ps = get_input_partial_shape(0);
+    const auto& bbox_deltas_ps = get_input_partial_shape(1);
+    const auto& image_shape_ps = get_input_partial_shape(2);
+    auto out_dimension = Dimension::dynamic();
 
+    if (class_probs_ps.is_static())
```

Comment:
Following `NODE_VALIDATION_CHECK` could be done also for `class_probs_ps` in case it is non-static.
Condition check: `class_probs.rank().compatible(4)`

Reply:
Sure, I'll add some testcases to cover dynamic inputs with wrong rank.